ops/testing.py

```diff
@@ -1759,6 +1789,104 @@ class NonAbsolutePathError(Exception):
     """
 
 
+class _MockStorage:
```

Comment:
I would rather not call this _Mock, as it doesn't act like a Mock (record, replay, check the calls, etc)
We specifically named it `_TestingModelBackend`, mabye `_TestingStorage` ? (Sorry I missed this for `_MockFilesystem`, maybe we should rename that one while we're here? Since it also isn't really act like a Mock). At the very least Fake would be  better than Mock for terminology here.


I would definitely like to see a docstring on various bits. Like what is 'location' vs what is 'src', etc.
I believe I know the answer today, but it won't be obvious in another month. :)

Also defining that `src` will be created if it doesn't exist is a pretty important point.